pkg/nodeconfig/nodeconfig.go

```diff
@@ -288,6 +304,186 @@ func (nc *nodeConfig) Configure() error {
 	return err
 }
 
+// CreateBootstrapFiles creates all prerequisite files on the node required to start kubelet using latest ignition spec
+func (nc *nodeConfig) createBootstrapFiles() error {
+	if err := nc.createFilesFromIgnition(); err != nil {
+		return err
+	}
+
+	if err := nc.createBootstrapKubeconfig(); err != nil {
+		return err
+	}
+
+	clusterDNS, err := cluster.GetDNS(nc.clusterServiceCIDR)
+	if err != nil {
+		return err
+	}
+	if err := nc.createKubeletConf(clusterDNS); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createFilesFromIgnition creates any files it can from ignition: kubelet CA cert, cloud-config file
+func (nc *nodeConfig) createFilesFromIgnition() error {
+	// For each new file in the ignition file check if is a file we are interested in, if so, decode it
+	// and write the contents to a temporary destination path
+	filesToTranslate := map[string]struct{}{
+		ignition.KubeletCACertPath: {},
+	}
+
+	ign, err := ignition.New(nc.client)
+	if err != nil {
+		return err
+	}
+	kubeletArgs, err := ign.GetKubeletArgs()
+	if err != nil {
+		return err
+	}
+	if _, ok := kubeletArgs[ignition.CloudConfigOption]; ok {
+		filesToTranslate[ignition.CloudConfigPath] = struct{}{}
+	}
+
+	for _, ignFile := range ign.Config.Storage.Files {
+		if _, ok := filesToTranslate[ignFile.Node.Path]; ok {
+			if ignFile.Contents.Source == nil {
+				return errors.Errorf("could not process %s: File is empty", ignFile.Node.Path)
+			}
+			contents, err := dataurl.DecodeString(*ignFile.Contents.Source)
```

Comment:
It looks like this library is being used within the coreos code so i guess its okay... but it feels not great using a random library like this.

Reply:
+1. 

>It looks like this library is being used within the coreos code

That doesn't automatically make it a good pattern. I suggest looking for an alternative if available.

Reply:
Moved to using `url.QueryUnescape`, looks like that does the trick

Reply:
@aravindhp @sebsoto After some testing, I cannot find an alternative standard library to get clean file contents. There's issues translating cloud.config when using `net/url`.

Using the `vincent-petithory/dataurl` library works for all files embedded in ignition so I lean towards using it. The only other alternative is some hacky string manipulation.

Reply:
> There's issues translating cloud.config when using net/url

What is the issue? I suggest opening a bug.

>Using the vincent-petithory/dataurl library works for all files embedded in ignition so I lean towards using it.

I am fine with that approach for now.

Reply:
It is not stripping the leading `data:,` tag of the contents which are in [data url scheme](https://www.rfc-editor.org/rfc/rfc2397.html). But it does strip this when parsing kubelet.conf contents. I'll look into opening a bug